roqoqo-qryd/src/tweezer_devices.rs

```diff
@@ -564,12 +600,20 @@ impl TweezerDevice {
         gate_time: f64,
         layout_name: Option<String>,
     ) -> Result<(), RoqoqoBackendError> {
-        self.qubit_to_tweezer = None;
+        if !ALLOWED_NATIVE_GATES.contains(&hqslang) {
+            return Err(RoqoqoBackendError::GenericError {
+                msg: format!(
+                    "Error setting the gate time of a multi-qubit gate. Gate {} is not supported.",
+                    hqslang
+                ),
+            });
+        }
         let layout_name = layout_name
             .or_else(|| self.current_layout.as_ref().map(|s| s.to_string()))
             .ok_or_else(|| RoqoqoBackendError::GenericError {
                 msg: "No layout name provided and no current layout set.".to_string(),
             })?;
+        self.qubit_to_tweezer = None;
```

Comment:
How come this line has been added in each of the functions in this file?

Reply:
It was already there, I moved it after the new error check as well as the layout getter to avoid modifying the internal data structure in case an error occurs. Like, if the user input a wrong layout name, the `qubit_to_tweezer` would have been unset anyway. This avoids that.